transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -113,7 +112,7 @@ final MessageSizeEstimator.Handle estimatorHandle() {
     }
 
     final Object touch(Object msg, AbstractChannelHandlerContext next) {
-        return touch ? ReferenceCountUtil.touch(msg, next) : msg;
+        return ResourceLeakDetector.isEnabled() ? ReferenceCountUtil.touch(msg, next) : msg;
```

Comment:
I'm thinking it's probably a good idea for pipelines, which can be around for some time, to respect online leak detection level changes.